blockchain/chain/src/store/chain_store.rs

```diff
@@ -107,19 +108,27 @@ where
             tipset_tracker: TipsetTracker::new(db.clone(), chain_config),
             db,
             ts_cache,
-            heaviest: Default::default(),
+            heaviest: Mutex::new(genesis_ts.clone()),
         };
 
         // Result intentionally ignored, doesn't matter if heaviest doesn't exist in store yet
         let _ = cs.load_heaviest_tipset();
 
-        cs
+        if cs.blockstore().read(HEAD_KEY)?.is_none() {
+            cs.set_genesis(&genesis_ts.blocks()[0])?;
+        }
+
+        if cs.blockstore().read(GENESIS_KEY)?.is_none() {
+            cs.set_heaviest_tipset(genesis_ts)?;
+        }
```

Comment:
The `HEAD_KEY` is used for the heaviest tipset. The `GENESIS_KEY` is used for the genesis block.

Additionally, it's easier just to always write the genesis key to the database.

Reply:
Made this behaviour default